numba/core/config.py

```diff
@@ -343,6 +343,11 @@ def avx_default():
         # gdb binary location
         GDB_BINARY = _readenv("NUMBA_GDB_BINARY", str, '/usr/bin/gdb')
 
+        # Casting and constant reference types
+        LOWER_CONSTANT_RETURNS_NEW_REFS = _readenv(
+            "NUMBA_LOWER_CONSTANT_RETURNS_NEW_REFS", int, 1)
+        CAST_RETURNS_NEW_REFS = _readenv("NUMBA_CAST_RETURNS_NEW_REFS", int, 1)
```

Comment:
I can see that these env-vars are useful during development, but I don't think we should give user the choice for something so dependent on internal details.

Reply:
Agreed, but I've seen that there are other projects using `context.cast` over on #5214. Probably would make the transition easier for them if they got at least a pre-release with switchable behaviour?
I'll implement this (and remark above) in a separate branch for now.